pkg/virt-controller/watch/vm.go

```diff
@@ -2941,21 +3061,21 @@ func (c *VMController) sync(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachin
 			}
 		}
 
-		if lastSeenVMSpec == nil || !equality.Semantic.DeepEqual(lastSeenVMSpec.Template.Spec.Domain.CPU, vm.Spec.Template.Spec.Domain.CPU) {
+		if c.clusterConfig.IsVMRolloutStrategyLiveUpdate() && !restartRequired && !conditionManager.HasCondition(vm, virtv1.VirtualMachineRestartRequired) {
```

Comment:
I am not sure if we need both `!restartRequired` and `!conditionManager.HasCondition(vm, virtv1.VirtualMachineRestartRequired)`

Reply:
That was just to avoid iterating over all the conditions when we know we just set restartRequired. I'm fine dropping it, please confirm.

Reply:
I am okay with keeping it, it should not matter.